data/model/autoprune.py

```diff
@@ -218,3 +232,199 @@ def namespace_has_autoprune_task(namespace_id):
+                    "autoprune_tag_delete",
+                    namespace.username,
+                    repository=repo,
+                    metadata={
+                        "performer": "autoprune worker",
+                        "namespace": namespace.username,
+                        "repo": repo.name,
+                        "tag": tag.name,
+                    },
+                )
+            except Exception as err:
+                raise Exception(
+                    f"Error deleting tag with name: {tag.name} with repository id: {repo.id} with error as: {str(err)}"
+                )
+
+
+def execute_policy_on_repo(policy, repo_id, namespace_id, tag_page_limit=100):
+    policy_to_func_map = {
+        AutoPruneMethod.NUMBER_OF_TAGS.value: prune_repo_by_number_of_tags,
+        AutoPruneMethod.CREATION_DATE.value: prune_repo_by_creation_date,
+    }
+
+    if policy_to_func_map.get(policy.method, None) is None:
+        raise KeyError("Unsupported policy provided", policy.method)
+
+    namespace = user.get_namespace_user_by_user_id(namespace_id)
+    repo = repository.lookup_repository(repo_id)
+
+    logger.info("Executing autoprune policy: %s on repo: %s", policy.method, repo.name)
+
+    policy_to_func_map[policy.method](repo, policy.config, namespace, tag_page_limit)
+
+
+def execute_policies_for_repo(policies, repo, namespace_id, tag_page_limit=100):
+    list(
+        map(
+            lambda policy: execute_policy_on_repo(policy, repo, namespace_id, tag_page_limit),
+            policies,
+        )
+    )
+
+
+def get_paginated_repositories_for_namespace(namespace_id, page_token=None, page_size=50):
+    try:
+        query = Repository.select(Repository.name, Repository.id,).where(
+            Repository.state != RepositoryState.MARKED_FOR_DELETION,
+            Repository.namespace_user == namespace_id,
+        )
+        repos, next_page_token = modelutil.paginate(
+            query,
+            Repository,
+            page_token=page_token,
+            limit=page_size,
+        )
+        return repos, next_page_token
+    except Exception as err:
+        raise Exception(
+            f"Error fetching paginated repositories for namespace id: {namespace_id} with error as: {str(err)}"
+        )
+
+
+def execute_namespace_polices(policies, namespace_id, repository_page_limit=50, tag_page_limit=100):
+    if not policies:
+        return
+    page_token = None
+
+    while True:
+        repos, page_token = get_paginated_repositories_for_namespace(
+            namespace_id, page_token, repository_page_limit
+        )
+        repo_list = [row for row in repos]
+
+        # When implementing repo policies, fetch repo policies and add it to the policies list here
+        list(
+            map(
+                lambda repo: execute_policies_for_repo(
+                    policies, repo, namespace_id, tag_page_limit
+                ),
+                repo_list,
+            )
+        )
```

Comment:
Why do you use `list(map(...))`?

```suggestion
        for repo in repo_list:
            execute_policies_for_repo(
                policies, repo, namespace_id, tag_page_limit
            )
```

Reply:
yeah we don't need that. I removed it.